### ISSUE TITLE
Assertion failure with v30 on Arch Linux

### ISSUE THREAD

**Issue by openglfreak** [bus1/dbus-broker#288]
dbus-broker 30 on Arch Linux (testing repo) is crashing for me:
```
May 10 23:22:51 ARKON at-spi-bus-launcher[1621]: dbus-broker: ../dbus-broker-30/src/util/user.c:195: user_free: Assertion `c_assert_result && "c_rbtree_is_empty(&user->usage_tree)"' failed.
```

Version 29 works fine.

coredump (from coredumpctl):
```
           PID: 888 (dbus-broker)
           UID: 81 (dbus)
           GID: 81 (dbus)
        Signal: 6 (ABRT)
     Timestamp: Tue 2022-05-10 23:23:00 CEST (3h 28min ago)
  Command Line: dbus-broker --log 4 --controller 9 --machine-id 19c070bce1e54fb3b24cabfceb60ac78 --max-bytes 536870912 --max-fds 4096 --max-matches 16384 --audit
    Executable: /usr/bin/dbus-broker
 Control Group: /system.slice/dbus-broker.service
          Unit: dbus-broker.service
         Slice: system.slice
       Boot ID: 22f742498a6a4dd29d42c144aa829de2
    Machine ID: 19c070bce1e54fb3b24cabfceb60ac78
      Hostname: ARKON
       Storage: /var/lib/systemd/coredump/core.dbus-broker.81.22f742498a6a4dd29d42c144aa829de2.888.1652217780000000.zst (present)
     Disk Size: 136.4K
       Message: Process 888 (dbus-broker) of user 81 dumped core.
                
                Module linux-vdso.so.1 with build-id 295910d5d8e8f3c9a9cf2c94f9b0fb86c4bc0af9
                Module ld-linux-x86-64.so.2 with build-id dd32d87cd4618fcf553fc02af302f9211b6ac769
                Module libc.so.6 with build-id d7c64b142d8d119e3e957416c63ee34c99ad99ef
                Module libgcc_s.so.1 with build-id 63dfd9229f32a325dc1c95d35bf9106fa9ed1031
                Module libcap-ng.so.0 with build-id 20a18041c641e46544872a81af23e40f1190a637
                Module libaudit.so.1 with build-id 3be019dda8c3bd355cbb170514a0e7f13875d36d
                Module libm.so.6 with build-id db56f3adc7024999dde66f49f0d657dd2d9bb767
                Module dbus-broker with build-id 6f5c864b69d83a01d7c2f4a2d4dd0850de2fbc2d
                Stack trace of thread 888:
                #0  0x00007fc9b489220c __pthread_kill_implementation (libc.so.6 + 0x9220c)
                #1  0x00007fc9b4842838 raise (libc.so.6 + 0x42838)
                #2  0x00007fc9b482c535 abort (libc.so.6 + 0x2c535)
                #3  0x00007fc9b482c45c __assert_fail_base.cold (libc.so.6 + 0x2c45c)
                #4  0x00007fc9b483b366 __assert_fail (libc.so.6 + 0x3b366)
                #5  0x000055c528dcd057 n/a (dbus-broker + 0x21057)
                #6  0x000055c528dc0412 n/a (dbus-broker + 0x14412)
                #7  0x000055c528dc444c n/a (dbus-broker + 0x1844c)
                #8  0x000055c528db1ee3 n/a (dbus-broker + 0x5ee3)
                #9  0x000055c528db010c n/a (dbus-broker + 0x410c)
                #10 0x00007fc9b482d290 __libc_start_call_main (libc.so.6 + 0x2d290)
                #11 0x00007fc9b482d34a __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x2d34a)
                #12 0x000055c528db0b45 n/a (dbus-broker + 0x4b45)
                ELF object binary architecture: AMD x86-64
```
[dbus-broker.dump.gz](https://github.com/bus1/dbus-broker/files/8665791/dbus-broker.dump.gz)

**Comment by Mel34** [comment]
+1

**Comment by dvdhrm** [comment]
Thanks a lot for the report! I pushed a possible fix just now: 608b259e25ef1348b9e4a8e022c35b5c68d3df98

**Comment by openglfreak** [comment]
Thank you! That fixes it
